crates/adapters/src/format/json/output.rs

```diff
@@ -129,6 +161,11 @@ impl JsonEncoder {
         let value_schema_str = build_value_schema(&config, schema);
         let key_schema_str = build_key_schema(&config, schema);
 
+        let key_fields = config
+            .key_fields
+            .as_ref()
+            .map(|fields| HashSet::from_iter(fields.iter().map(|f| canonical_identifier(f))));
```

Comment:
I guess this takes care of casing

Reply:
we should really wrap these things in a type that implements eq in a way that does the right thing